### PR TITLE
Add TLVs: REST server, URI scheme, and record packets

### PR DESCRIPTION
**Pull request by rizard**
Reviewer: @wilmo119 @kenchiang 

I'm including a uint8_t for record_packets even though it's just a boolean... just in case we support more than on/off in the future.

### CODE REVIEW

**Comment by bsn-abat** [comment]
Find artifact changes from this pull request at https://github.com/floodlight/loxigen-artifacts/tree/pr-551-jenkins-loxigen_pull_req_artifacts-101

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req_artifacts/101/

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req_docker/118/

File: openflow_input/bsn_tlv [hunk 1]
```diff
@@ -1077,8 +1077,25 @@ struct of_bsn_tlv_force_link_up : of_bsn_tlv {
...
 struct of_bsn_tlv_timestamp : of_bsn_tlv {
     uint16_t type == 154;
     uint16_t length;
     uint64_t value;
 };
+
+struct of_bsn_tlv_record_packets : of_bsn_tlv {
+    uint16_t type == 155;
+    uint16_t length;
+    uint8_t value;
```
Comment (kenchiang) [inline openflow_input/bsn_tlv]: do you foresee a uint8_t is sufficiently large, or would a uint32_t be more appropriate (say for flags)?

Comment (rizard) [inline openflow_input/bsn_tlv]: uint8_t should be plenty, but I can change to a uint32_t just to be safe if you'd like

Comment (andi-bigswitch) [inline openflow_input/bsn_tlv]: Given the overhead of the TLV structure I'd say let's go with 32 bits...

Comment (rizard) [inline openflow_input/bsn_tlv]: It's now 32 bits

**Comment by bsn-abat** [comment]
Find artifact changes from this pull request at https://github.com/floodlight/loxigen-artifacts/tree/pr-551-jenkins-loxigen_pull_req_artifacts-102

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req_artifacts/102/

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req_docker/119/

**Comment by bsn-abat** [comment]
Find artifact changes from this pull request at https://github.com/floodlight/loxigen-artifacts/tree/pr-551-jenkins-loxigen_pull_req_artifacts-104

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req_artifacts/104/

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/loxigen_pull_req_docker/121/

**Comment by rizard** [comment]
Does this look good?

**Comment by rizard** [comment]
SELF-ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vss1/abat/2017.05.03.0010-m.loxigen.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://vss1/abat/2017.05.03.0010-m.loxigen.master/abat.log
/cc

**Pull request merged**